src/trail.js

```diff
@@ -1,29 +1,21 @@
-/**
+/*
  * Copyright (c) 2016-present, IBM Research
  * Licensed under The MIT License [see LICENSE for details]
  */
 
 'use strict'
 
-const shortid = require('shortid');
-const Types = require('./types');
-const HKEntity = require('./hkentity');
-const Connector = require('./connector');
-const Link = require('./link');
-const RoleTypes = require('./roletypes');
+const {List, Item}    = require('linked-list');
 
-const CONNECTOR_NAME = 'occurs';
+// const shortid   = require('shortid');
+const Types     = require('./types');
+const HKEntity  = require('./hkentity');
 
-const vConnector = new Connector(CONNECTOR_NAME, 'f');
-vConnector.addRole('sub', RoleTypes.SUBJECT);
-vConnector.addRole('obj', RoleTypes.OBJECT);
 
-class Trail extends HKEntity
-{
-  constructor(id, parent)
+function Trail (id, actions, parent)
```

Comment:
Couldn't this function be a class?

Reply:
Yes, I forgot to update, but I'll do it now